### PR TITLE
[compatibility] Restore some of compatibility

### PR DESCRIPTION
**Pull request by pkriens**
- Added FixedIndexedRepo that was removed in R4, now extends OSGiRepository which was designed plug compatible. Gives a warning
– Deprecated bnd annotations
– Better diagnostics for the Service-Component header

Signed-off-by: Peter Kriens <peter.kriens@aqute.biz>

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/make/component/ServiceComponent.java [hunk 1]
```diff
@@ -119,9 +121,73 @@ private void componentEntry(Map<String, Map<String, String>> serviceComponents,
...
+			if (where != null) {
+				where.set(warning);
+			}
+
+			warning.reference(
+				"https://osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#d0e36857");
+
+			TypeRef componentClassRef = analyzer.getTypeRefFromFQN("aQute.bnd.annotation.component.Component");
+			Clazz componentClass = analyzer.findClass(componentClassRef);
+			if (componentClass == null) {
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/make/component/ServiceComponent.java]: But couldn't the bundle being built include a package from a jar on the classpath? This does not require the Bnd DS annotations to be on the classpath. They are only needed on the classpath to compile source not to include a package already compiled.

File: biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java [hunk 2]
```diff
@@ -47,66 +49,80 @@ protected void setUp() throws Exception {
...
-						}
-
-						@Override
-						public boolean isCanceled() {
-							return false;
-						}
-					};
-				}
-			});
+			assertTrue(testRepo(r).check());
```
Comment (bjhargrave) [inline biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java]: `testRepo` returns a `Processor` that should be closed.

```java
try (Processor p = testRepo(r)) {
  assertTrue(p.check());
}
```

File: biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java [hunk 3]
```diff
@@ -47,66 +49,80 @@ protected void setUp() throws Exception {
...
-
-			r.refresh();
-			File file3 = r.get("dummybundle", new Version("0"), null);
-			assertNotNull(file3);
-			// second one should not have downloaded
-			assertEquals(2, tasks.get());
+	@SuppressWarnings("deprecation")
+	public void testCompatibilityWithFixedIndexedRepo() throws Exception {
+		try (FixedIndexedRepo r = new FixedIndexedRepo();) {
+			assertTrue(testRepo(r)
```
Comment (bjhargrave) [inline biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java]: Also need to close Processor returned by testRepo.

File: docs/_heads/include_resource.md [hunk 4]
```diff
@@ -3,33 +3,8 @@ layout: default
 class: Header
```
Comment (bjhargrave) [inline docs/_heads/include_resource.md]: Isn't this file already changed by a merged PR?

File: docs/_instructions/includeresource.md [hunk 5]
```diff
@@ -1,282 +1,35 @@
 ---
```
Comment (bjhargrave) [inline docs/_instructions/includeresource.md]: Isn't this file already changed by a merged PR?

File: biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java (hunk 2 above)
Comment (pkriens) [reply biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java]: this is a test case and free processor that will be gc'ed

File: docs/_heads/include_resource.md (hunk 4 above)
Comment (pkriens) [reply docs/_heads/include_resource.md]: will be changed more complete soon

File: biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java (hunk 3 above)
Comment (pkriens) [reply biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java]: as before

File: biz.aQute.bndlib/src/aQute/bnd/make/component/ServiceComponent.java (hunk 1 above)
Comment (pkriens) [reply biz.aQute.bndlib/src/aQute/bnd/make/component/ServiceComponent.java]: A bit of an edge case I guess you're right. I can fix that later.

**Pull request merged**